server/controllers/shoppingList.js

```diff
@@ -157,22 +158,30 @@ const getListData = (req, resp) => {
       { purchaserIds: req.user._id }
     ]
   })
-    .then(doc => {
-      if (!doc) {
-        return resp.status(404).send({ message: 'List data not found.' });
+    .exec()
+    .then(
+      doc => {
+        if (!doc) {
+          throw new ResponseError('List data not found.', 404);
+        }
+        list = doc;
+        const { cohortId } = list;
+        if (cohortId) {
+          return Cohort.findOne({ _id: cohortId })
+            .exec()
+            .then(cohort => {
+              if (!cohort || cohort.isArchived) {
+                throw new ResponseError('List data not found.', 404);
+              }
+            });
+        }
+      },
+      err => {
+        if (err.name === 'CastError') {
```

Comment:
can you explain here a little bit?
Wy you checking is err.name is 'CastError' and not just checking if err exist?

Reply:
This code will be removed. It's unclear and not proper. It was added to handle this specific error which occurs when id is not valid (e.g. too short). Before that, in this situation catch method was handling this error, and was returning 400. This solution handle this specific error and return 404 and frontend app can react on it.
We should think about all actions which user can perform and create proper scenariouses of handling errors and when and how show to user proper message.